pkg/controller/cluster/main-controller.go

```diff
@@ -352,31 +353,32 @@ func (c *Controller) syncHandler(key string) error {
 			runtime.HandleError(fmt.Errorf("Tenant '%s' in work queue no longer exists", key))
 			return nil
 		}
-		return err
+		return nil
 	}
-
 	// Set any required default values and init Global variables
 	nsName := types.NamespacedName{Namespace: namespace, Name: name}
-	mi.EnsureDefaults()
+	mi = mi.EnsureDefaults()
 	miniov1.InitGlobals(mi)
-
 	// Validate the MinIO Instance
 	if err = mi.Validate(); err != nil {
 		klog.V(2).Infof(err.Error())
 		var err2 error
-		mi, err2 = c.updateTenantStatus(ctx, mi, err.Error(), 0)
-		if err2 != nil {
+		if mi, err2 = c.updateTenantStatus(ctx, mi, err.Error(), 0); err2 != nil {
 			klog.V(2).Infof(err2.Error())
 		}
-		return err
+		return nil
```

Comment:
The processNextWorkItem() loops logs and error at log level error. Why skip returning the error? Here and below.

Reply:
It's causing the Tenants that go into an error state to loop endlessly, if nothing was changed to fix this problem, there's no sense in re-processing the objects. The error is still returned during the zones loop